### PR TITLE
FEM: export of netgen/gmsh mesh into fenics xml format

### PR DESCRIPTION
**Pull request by joha2**
With this PR you are now able to export meshes from netgen or gmsh into fenics xml files.
All comments or critics are welcome.

### CODE REVIEW

**Comment by berndhahnebach** [comment]
For such a change we should only use one commit, means squash them. 

entry in src/Mod/Fem/CMakeLists.txt is missing.

bernd

File: src/Mod/Fem/importFenicsMesh.py [hunk 1]
```diff
@@ -77,116 +77,111 @@ def export(objectslist, filename):
...
-		nodeindices = fem_mesh_obj.FemMesh.getElementNodes(fc_volume_ind)
-
-		cell_args = {}
-		for (vi, ni) in enumerate(nodeindices):
-			cell_args["v" + str(vi)] = str(ni - 1)
-		# generate as many v entries in dict as nodes are listed in cell (works only for first order elements)
-
-		etree.SubElement(cells, cellname_fenics, index=str(fen_ind), **cell_args)
-
-	data = etree.SubElement(meshchild, "data")
```
Comment (berndhahnebach) [inline src/Mod/Fem/importFenicsMesh.py]: Is it intended to delete the data = ?

**Comment by berndhahnebach** [comment]
some comment:

https://github.com/FreeCAD/FreeCAD/pull/559/commits/1555e86efb61542ce7fc691d0a2a6ab75818ffa5?diff=unified#diff-5afa621ece7f73dc78a554aff646ed6cL188

**Comment by berndhahnebach** [comment]
In FEM we mostly try to stick to flake8 for python coding. Did you test your module with this?

Bernd

**Comment by joha2** [comment]
Hi Bernd,
thanks for reviewing. First of all I will close this PR, as you requested and submit it via your dev branch. To your other points:
* The deletion of data was intendend since it is never used (pylint complained). But the file structure has such a data line
* No, I did not check via flake8, but I will do :-)
* How to stash commits? I am fairly new to git :-(
Best wishes
Johannes,
PR: closed

**Comment by berndhahnebach** [comment]
git rebase -i HEAD 2
an editor will open and you see the options. s means squash, but be careful. Test first with a new branch. 

Ahh I could the squashing for you too. 
bernd

**Comment by bblacey** [comment]
Another way to accomplish the same if you are unsure of rebase -i...
```
git reset HEAD~1        #pop last commit but keep changes local
git add <files>            #stage files from commit that was reset
git commit --amend  #amend original commit to include changes from commit reset
git push -f origin fenics_export #push back to GitHub.  You need -f to force an update of the rewritten history
```
Providing in case this is easier to understand than rebase -i but that is definitely a command you should learn.

No need to close the PR and re-open - just update your branch and the PR will pick up the changes.

**Comment by joha2** [comment]
Bernd and blacey thanks for your hints about git. I will try them out. @bblacey I closed the PR due to the decision to incorporate it into the femdev branch of bernd after fixing the issues.
Best regards
Johannes

**Pull request closed without merging**